src/textual/widgets/_data_table.py

```diff
@@ -184,8 +157,10 @@ class DataTable(ScrollView, Generic[CellType], can_focus=True):
     show_cursor = Reactive(True)
     cursor_type = Reactive(CELL)
 
-    cursor_cell: Reactive[Coord] = Reactive(Coord(0, 0), repaint=False)
-    hover_cell: Reactive[Coord] = Reactive(Coord(0, 0), repaint=False)
+    cursor_cell: Reactive[Coordinate] = Reactive(
+        Coordinate(0, 0), repaint=False, always_update=True
```

Comment:
`always_update` is required as sometimes we need to force validation via `self.cursor_cell = self.cursor_cell`.